### PR TITLE
add test for solving with a partial starting point

### PR DESCRIPTION
**Pull request by mlubin**
Ref https://github.com/JuliaOpt/Ipopt.jl/issues/155

### CODE REVIEW

File: src/Test/contlinear.jl [hunk 1]
```diff
@@ -1459,6 +1459,41 @@ function linear15test(model::MOI.ModelLike, config::TestConfig)
     end
 end
 
+# Exclude this test if the solver isn't intended to support incomplete starting
```
Comment (blegat) [inline src/Test/contlinear.jl]: Where is the starting point set ?

Comment (mlubin) [reply src/Test/contlinear.jl]: Good catch! :sweat_smile:

**Comment by blegat** [comment]
We won't be able to include it in MOI v0.8.2 as it will make almost all solver tests fail or we will have to force it on METADATA.
It would be nice if contlinear didn't use primal start do that all solvers can pass all the tests.
To me the new test feel more like a unit test.

**Comment by mlubin** [comment]
METADATA isn't currently running reverse dependency tests, so we're not blocked by that.

Yes, I'd guess most solvers fail this test with an unhelpful error message, but that's a pretty significant usability issue and likely regression form JuMP 0.18 that we should be aware of and try to address. We can either add the test here so we're forced to deal with it, or open issues on all of the solver wrappers.

**Comment by blegat** [comment]
What I mean is that some solvers do not even support starting points.
For solvers that support it but wrappers fail when it is partially set, I agree that their tests should fail.
I am concerned that this prevents solvers not supporting starting points to pass the tests. Maybe a `config` parameter could help

**Comment by mlubin** [comment]
> I am concerned that this prevents solvers not supporting starting points to pass the tests. 

They can always exclude the test, but should have a good reason to do so. Erroring when the solver does not support warm starts is a regression from JuMP 0.18. This was previously a warning: https://github.com/JuliaOpt/JuMP.jl/blob/d8f526d48f9c79f6aa27ef45aea381b4b7e63f32/src/solvers.jl#L413.

**Comment by mlubin** [comment]
> Maybe we should implement the warning in MOI then instead of an ArgumentError.

That makes sense.

**Comment by blegat** [comment]
Then we don't have to exclude this test from `test/Utilities/...`, `test/bridge.jl`, ...

**Pull request merged**